src/apppermission.cpp

```diff
@@ -108,6 +108,9 @@ void AppPermission::flip(const QString& appID) {
 
   int index = m_applist.indexOf(AppDescription(appID));
   emit dataChanged(createIndex(index, 0), createIndex(index, 0));
+
+  beginResetModel();
```

Comment:
@MattLichtenstein why do we need this?

Reply:
I was using this to reset the app permission list model so that the UI would update without having to reload the list component (aka popping the view off the stack and re-pushing it). Do you think there is there a better way to do this?

Reply:
OK! we can work on this as a follow up.